leshan-server-demo/webapp2/src/views/Client.vue

```diff
@@ -0,0 +1,125 @@
+<template>
+  <v-row class="fill-height" no-gutters>
+    <v-col cols="12" md="2">
+      <!-- registration info -->
+      <v-sheet color="grey lighten-5" class="pa-4" width="100%">
+        <div>
+          <h3>{{ $route.params.endpoint }}</h3>
+          <div v-if="registration">
+            <div>Reg. ID: {{ registration.registrationId }}</div>
+            <div>Using LWM2M v{{ registration.lwM2mVersion }}</div>
+            <div>
+              Updated: 
+              {{ registration.lastUpdate | moment("MMM D, h:mm:ss A") }}
+            </div>
+          </div>
+        </div>
+        <v-divider />
+        <client-setting />
+      </v-sheet>
+
+      <v-divider />
+
+      <!-- object selector -->
+      <object-selector :objects="objects" />
+
+      <v-divider></v-divider>
+    </v-col>
+
+    <v-col no-gutters cols="12" md="10">
+      <!-- object viewer -->
+      <router-view
+        :object="objects.find((o) => o.id == $route.params.objectid)"
+        :instances="instances"
+      ></router-view>
+    </v-col>
+  </v-row>
+</template>
+
+<script>
+import ClientSetting from "../components/ClientSetting.vue";
+import ObjectSelector from "../components/ObjectSelector.vue";
+
+// get models for this endpoint
+export default {
+  components: { ObjectSelector, ClientSetting },
+  name: "Client",
+  data: () => ({
+    registration: null,
+    objects: [],
+  }),
+  methods: {
+    updateModels: function() {
+      this.axios
+        .get(
+          "api/objectspecs/" + encodeURIComponent(this.$route.params.endpoint)
+        )
+        .then((response) => (this.objects = response.data))
+        .catch((error) => console.log(error));
+    },
+  },
+  computed: {
+    // a computed getter
+    instances: function() {
+      // should probably be returned by REST API
+      // TODO we need to handle rootpath
+      let instances = [];
+      if (this.registration) {
+        this.registration.objectLinks.forEach((o) => {
+          let ids = o.url.split("/");
+          if (ids.length === 3 && ids[1] == this.$route.params.objectid)
+            instances.push({ id: ids[2] });
+        });
+      }
+      return instances;
+    },
+  },
+  watch: {
+    /*$route(to, from) {
+      console.log("route", to, from);
+    },*/
```

Comment:
Commented code

Reply:
This is where we should handle url changes, we probably need to to that.
But not a priority so I guess I will keep it and add a TODO